src/Reflex/Dom/Widget/Input.hs

```diff
@@ -212,6 +214,110 @@ dropdown k0 options (DropdownConfig setK attrs) = do
+                 , _buttonGroup_element  :: El t
+                 , _buttonGroup_children :: Dynamic t (Map.Map Int (El t))
+                 }
+
+data ButtonGroupConfig t a
+   = ButtonGroupConfig { _buttonGroupConfig_parentType   :: String
+                       , _buttonGroupConfig_parentTag    :: String
+                       , _buttonGroupConfig_initialValue :: Maybe a
+                       , _buttonGroupConfig_setValue     :: Event t (Maybe a)
+                       , _buttonGroupConfig_attributes   :: Dynamic t (Map String String)
+                       }
+
+instance Reflex t => Default (ButtonGroupConfig t a) where
+  def = ButtonGroupConfig { _buttonGroupConfig_parentType   = ""
+                          , _buttonGroupConfig_parentTag    = "div"
+                          , _buttonGroupConfig_initialValue = Nothing
+                          , _buttonGroupConfig_setValue     = never
+                          , _buttonGroupConfig_attributes   = constDyn mempty
+                          }
+
+buttonGroup :: (MonadWidget t m, Eq a, Show a)
+            => (Maybe Int -> Dynamic t a -> Dynamic t Bool -> m (Event t (), El t))
+            -> Dynamic t (Map.Map Int a)
+            -> ButtonGroupConfig t a
+            -> m (ButtonGroup t a)
+buttonGroup drawBtn dynBtns (ButtonGroupConfig pType pTag iVal setV dAtts) = do
+  dAtts' <- mapDyn (Map.insert "type" pType) dAtts
+  (parent, (dynV, internV, child)) <- elDynAttr' pTag dAtts' $ mdo
+    pb <- getPostBuild
+    let externSet = attachWith revLookup (current dynBtns) setV
+        initSet   = attachWith revLookup (current dynBtns) (iVal <$ pb)
+        internSet = leftmost [initSet, clickSelEvts]
+        internVal = attachWith (\m k -> k >>= flip Map.lookup m)
+                              (current dynBtns)
+                              internSet
+    dynK     <- holdDyn Nothing $ leftmost [internSet, externSet]
+    dynBtns' <- mapDyn (Map.mapKeys Just) dynBtns
+    (clickSelEvts, children) <- selectViewListWithKey_' dynK dynBtns' drawBtn
+    dynSelV <- combineDyn (\k m -> k >>= flip Map.lookup m) dynK dynBtns
+    return (dynSelV, internVal, children)
+  return (ButtonGroup { _buttonGroup_value    = dynV
+                      , _buttonGroup_select   = internV
+                      , _buttonGroup_element  = parent
+                      , _buttonGroup_children = child
+                      })
+ 
+revLookup :: Eq a => Map.Map Int a -> Maybe a -> Maybe Int
+revLookup _ Nothing = Nothing
+revLookup m (Just v) = listToMaybe . Map.keys $ Map.filter (== v) m
+
+selectViewListWithKey_' :: forall t m a v.(MonadWidget t m)
+                        => Dynamic t (Maybe Int)
+                        -> Dynamic t (Map.Map (Maybe Int) v)
+                        -> (Maybe Int -> Dynamic t v -> Dynamic t Bool -> m (Event t a, El t))
+                        -> m (Event t (Maybe Int), Dynamic t (Map.Map Int (El t)))
+selectViewListWithKey_' selection vals mkChild = do
```

Comment:
Is it possible to either implement this in terms of one of the other select*list* functions or to implement them in terms of this?  Also, can we generalize the types here?